test/grpc_test.go

```diff
@@ -56,3 +59,75 @@ func TestGrpc(t *testing.T) {
 		t.Errorf("Expected 2 RRs in additional section, but got %d", len(d.Extra))
 	}
 }
+
+type grpcDnsServiceServer struct {
+	pb.UnimplementedDnsServiceServer
+}
+
+func (*grpcDnsServiceServer) Query(ctx context.Context, in *pb.DnsPacket) (*pb.DnsPacket, error) {
+	msg := &dns.Msg{}
+	msg.Unpack(in.GetMsg())
+	answer := new(dns.Msg)
+	answer.SetRcode(msg, dns.RcodeSuccess)
+	buf, _ := answer.Pack()
+	return &pb.DnsPacket{Msg: buf}, nil
+}
+
+func TestGrpcUnix(t *testing.T) {
+
+	fd := "/tmp/test745260595.grpc"
```

Comment:
Take a look at how [unixsock_test.go](https://go.dev/src/net/unixsock_test.go) does it.
[testUnixAddr](https://go.dev/src/net/mockserver_test.go) uses `os.MkdirTemp` to get a name that is unique

Reply:
ok